docs/core/versions/binding.md

```diff
@@ -0,0 +1,170 @@
+
+- 2.0.4 is required. No 2.0.* versions are installed. 2.2.2 is installed. 2.2.2 is used.
+- 2.0.5 is later installed. 2.0.5 will be used for subsequent application launches, not 2.2.2.
+- It's possible that 2.0.5 and 2.2.2 might behave differently, particularly for scenarios like serializing binary data.
+
+NuGet dependencies are treated differently. An application might be built for .NET Core 2.0 but run on .NET Core 2.1 due to this algorithm. There may be later versions of an application's NuGet dependencies that target .NET Core 2.1. These later versions aren't considered or installed. All NuGet dependencies are resolved as part of the publish operation and reside as assemblies (.dlls) in a flat directory structure. These assemblies should run on .NET Core 2.1 due to its compatibility promise for existing applications and binaries.
+
+Occasionally, ASP.NET packages are deployed via a web host rather than with a published application. In those cases, the web host should correctly configure their environments based on published guidance such that all supported ASP.NET applications run correctly.
+
+Developers should remember the following rules:
+
+- Runtimes are searched for in the following order (the first found is loaded):
+  - Highest patch version (check that the minimum runtime path version is satisfied, otherwise error)
+  - Highest minor version
+- If an acceptable runtime is not found, the application does not run and an error message explains why.
+
+## Publish a self-contained application
+
+You can publish an application as a **self-contained distribution**. This approach includes .NET Core. Self-contained distributions do not have a dependency on runtime environments. Runtime binding occurs at publishing time, not run-time.
+
+The publishing process will select the latest patch version of the given runtime family. For example, `dotnet publish` will select .NET Core 2.0.4 if it is the latest patch version in the .NET Core 2.0 runtime family, independent of the existence of earlier patch versions or minimum runtime patch version values. The publishing process differs from `dotnet build` runtime selection for two reasons:
+
+- The application doesn't specify a hosting environment.
+- The developer generates a final configuration for the application (that can't be practically modified). The best runtime choice is the latest installed runtime patch version. It's the same version used for development and it has the latest (per the machine) security and reliability fixes.
+
+The `runtimeframeworkversion` element will override the default version policy if that element is present in the project file.
+
+It is an error if the minimum version specified for an application is not satisfied. `dotnet publish` binds to latest runtime patch version (within a given major.minor version family). `dotnet publish` does not support the roll-forward semantics of `dotnet run`.
```

Comment:
Second sentence has already been stated. Third is problematic. It does support patch roll-forward, as already discussed. The rules are different, which if it's going to be stated should probably be in the first paragraph of the self-contained section. 

@dsplaisted can you confirm that at publish there is not runtime minor roll-forward?